src/custom/hooks/web3.ts

```diff
@@ -85,9 +85,18 @@ export function useEagerConnect() {
 
   // if the connection worked, wait until we get confirmation of that to flip the flag
   useEffect(() => {
+    let timeout: NodeJS.Timeout | undefined
+
     if (active) {
       setTried(true)
+    } else {
+      timeout = setTimeout(() => {
+        localStorage.removeItem(STORAGE_KEY_LAST_PROVIDER)
+        setTried(true)
+      }, WAITING_TIME_RECONNECT_LAST_PROVIDER)
     }
+
+    return () => timeout && clearTimeout(timeout)
```

Comment:
why do we clear the timeout? I understand we don't want to do the `setTried` but, if you clear it you won't do the `removeItem`

Reply:
The timeout is because if during the waiting time the provider changes its [`active`](https://github.com/gnosis/cowswap/tree/develop/src/custom/hooks/web3.ts#L100) sate, it would no longer need to remove the [`lastProvider`](https://github.com/gnosis/cowswap/tree/develop/src/custom/hooks/web3.ts#L66).

And ultimately `removeItem` will be executed by [handleBeforeUnload](https://github.com/gnosis/cowswap/tree/develop/src/custom/hooks/web3.ts#L25)